manifests/istio-control/istio-discovery/templates/deployment.yaml

```diff
@@ -295,20 +295,21 @@ spec:
         configMap:
           name: istio-validation
           optional: true
-      {{- end }}
-
-      - name: config-volume
-        configMap:
-          name: istio{{ .Values.version }}
+        {{- else }}
       - name: pilot-envoy-config
         configMap:
           name: pilot-envoy-config{{ .Values.version }}
-  {{- if .Values.global.controlPlaneSecurityEnabled}}
+        {{- if .Values.global.controlPlaneSecurityEnabled}}
       - name: istio-certs
         secret:
           secretName: istio.istio-pilot-service-account
           optional: true
-  {{- end }}
+        {{- end }}
+        {{- end }}
+
+      - name: config-volume
```

Comment:
what is this for? we already have the "istiod" one?

Reply:
No, that's the config map created by configmap.yaml, with the mesh config.

Needed in both pilot and istiod modes.

Reply:
why are we adding a volume if it isn't mounted though?

Reply:
nevermind it moved..